vars/assembleUpload.groovy

```diff
@@ -6,10 +6,9 @@
  * this file be licensed under the Apache-2.0 license or a
  * compatible open source license.
  */
-void call(Map args = [:]) {
-
-    lib = library(identifier: 'jenkins@main', retriever: legacySCM(scm))
 
+void call(Map args = [:]) {
+    lib = library(identifier: 'jenkins', retriever: legacySCM(scm))
```

Comment:
So basically just call the `jenkins` as identifier is enough?

Reply:
So I removed main @ for now. The documentation said (if I understand it right,) the leagacySCM plugin will use whatever is available. So in this case if the pipeline has clone jenkins@2.0.0 I believe it will use whatever is in vars folder.
If I run one of the tests that uses that, here is the ouput:https://github.com/opensearch-project/opensearch-build-libraries/blob/46f28cddcf2e7fb3e685663e2dd7ce815390da80/tests/jenkins/jobs/ArchiveAssembleUpload_Jenkinsfile.txt

Reply:
Thank you, @gaiksaya ..
What would you suggest to do at .defaultVersion parameter in helper.registerSharedLibrary at build repo?
I tried removing it, but looked like it could not be null ..

Here is the code :
        helper.registerSharedLibrary(
            library().name('jenkins')
                // .defaultVersion('1.0.0')
                .allowOverride(true)
                .implicit(true)
                .targetPath('vars')
                .retriever(gitSource('https://github.com/opensearch-project/opensearch-build-libraries.git'))
                .build()
        )

and here is the output 

TestWhileSourceScan > testWhileSourceScan FAILED
    java.lang.IllegalStateException: LibraryConfiguration is not properly initialized LibraryConfiguration{name='jenkins', defaultVersion='null', retriever=GitSource{sourceURL='https://github.com/opensearch-project/opensearch-build-libraries.git'}, implicit=true, 
allowOverride=true}
        at com.lesfurets.jenkins.unit.global.lib.LibraryConfiguration.validate(LibraryConfiguration.groovy:23)
        at com.lesfurets.jenkins.unit.global.lib.LibraryConfiguration$1.build(LibraryConfiguration.groovy:28)
        at com.lesfurets.jenkins.unit.global.lib.LibraryConfiguration$1$build.call(Unknown Source)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:130)
        at TestWhileSourceScan.setUp(TestWhiteSourceScan.groovy:28)

Thanks!

Reply:
Was that modernSCM or legacySCM?

Reply:
Hi, @gaiksaya ,, the groovy test job at build repo uses gitSource to refer to the remote lib at butild-lib repo. I believe it is the modenSCM as it is using the remote lib if I understand the document right ( v.s the legacySCM which is using the lib at local) , thanks!